query-engine/black-box-tests/tests/helpers/mod.rs

```diff
@@ -34,12 +34,30 @@ where
     Ok(())
 }
 
-/// Configures the query-engine binary to listen in given port and using the
-/// given dml string as the schema.
-pub(crate) fn query_engine_cmd(dml: &str, port: &str) -> process::Command {
+// Sequentially generates port numbers, which are potentially free. No checks
+// are performed to ensure that nobody actually occupies the port.
+fn generate_free_port() -> u32 {
+    static PORT_NUMBER: Mutex<u32> = Mutex::new(57582);
```

Comment:
There are more reliable ways to do that by starting a query-engine with a zero port number and then trying to find which port the OS gave us, but this sounds a bit overcomplicated for what we are trying to achieve here.

Reply:
On the same vibes as below, the easiest solution is the best here. This'll work just fine for our simple needs.